node/src/node_configurator/mod.rs

```diff
@@ -207,6 +217,35 @@ mod tests {
         )
     }
 
+    #[test]
+    fn data_directory_from_dir_opt_creates_new_folder_for_every_blockchain_platform() {
```

Comment:
The name should be improved. Notice that there is already a test testing these things but your name suggests that this test covers all cases. That's not right, this test is only about "creating_the_right_directory_hierarchy_for_user_requested_data_directory"

Reply:
renamed to `data_directory_from_context_creates_new_folder_for_every_blockchain_platform`